src/main/java/org/broadinstitute/hellbender/engine/filters/VariantIDsVariantFilter.java

```diff
@@ -23,15 +25,8 @@ public VariantIDsVariantFilter(Set<String> keepIDs) {
     @Override
     public boolean test(final VariantContext vc) {
         if (vc.getID().indexOf(';') > 0) {
-            String[] vc_ids = vc.getID().split(";");
-            for (String vc_id : vc_ids) {
-                if (includeIDs.contains(vc_id)) {
-                    return true;
-                }
-            }
-            return false;
-        } else {
-            return includeIDs.contains(vc.getID());
+            return Arrays.stream(vc.getID().split(";")).anyMatch(includeIDs::contains);
         }
+        return includeIDs.contains(vc.getID());
```

Comment:
You don't need this line or `if (vc.getID().indexOf(';') > 0)`.  If the field doesn't contain ";" `String.split(";")` still works; it returns a length-1 array.  So in fact the entire `test()` method can be the single line `return Arrays.stream(vc.getID().split(";")).anyMatch(includeIDs::contains)`.

Reply:
:+1:

Reply:
OK